### ISSUE TITLE
Wrong experiment associated with study_567

### ISSUE THREAD

**Issue by lwaldron** [waldronlab/BugSigDB#143]
https://bugsigdb.org/Study_567 presents a Chinese study of colorectal adenoma, created recently by user Jeshudy on May 30, 2022. But Experiment 1 is from a Russian study on Parkinson's disease created by user Fcuevas3 on Nov 29, 2021. Experiment 2 seems to actually belong to Study 567, created on May 31 by Jeshudy.

**Comment by lwaldron** [comment]
This misplaced experiment https://bugsigdb.org/Study_567/Experiment_1 seems to belong to the Russian study https://bugsigdb.org/Study_22 that has a reviewed Experiment 1 (https://bugsigdb.org/Study_22/Experiment_1) that seems correct, and to refer to the same real-life experiment as https://bugsigdb.org/Study_567/Experiment_1, only with some curation inconsistencies in the group naming and statistical test. 

The misplaced https://bugsigdb.org/Study_567/Experiment_1 is marked as "incomplete" - I'll look for some other incomplete pages to see if that is related to being misplaced.

**Comment by tosfos** [comment]
Copying from my email:
> It looks like the Study and Experiment were created together. Then someone deleted the Study (see [here](https://bugsigdb.org/w/index.php?title=Special:Log&page=Study+567)) without deleting the Experiment page, which is very easy to do. When a new Study page was then created, the Experiment 1 page was still in the wiki, which caused all the confusion.

We implemented a system to help find orphaned pages as happened here. You can see it [here](https://bugsigdb.org/w/index.php?title=Help:Cleanup&oldid=99217).

**Comment by tosfos** [comment]
Please let me know if we should clean up this particular Study.

**Comment by lwaldron** [comment]
Yes please, if you can. In other ones that don't have some correct and some incorrect experiments like #145, I think we could fix it on our own just by changing the PMID of the study.

**Comment by tosfos** [comment]
This can be closed

**Comment by lwaldron** [comment]
It looks like we can do the cleanup, but I’m not sure I understand how this
signature is listed as orphaned, but links back to what appears to be the
parent? Does it have to do with it being marked as Incomplete?

https://bugsigdb.org/Study_438/Experiment_4/Signature_1

-- 

Levi Waldron

Associate Professor

Department of Epidemiology and Biostatistics

CUNY Graduate School of Public Health and Health Policy

Institute for Implementation Science in Population Health

55 W 125th St, New York NY 10035

https://waldronlab.io

Join the microbiome Virtual International Forum: https://microbiome-vif.org

**Comment by lwaldron** [comment]
This seems to have happened again here for https://bugsigdb.org/Study_340. I feel like we need to lock down deleting studies because this is really hard to spot except by chance and I can't tell which user deleted this study. A new user is associated with the creation of this new study now associated with the wrong experiments. See
https://github.com/waldronlab/BugSigDBcuration/issues/23

**Comment by ftzohra22** [comment]
@lwaldron just to clarify I deleted the study associated with the experiment in https://bugsigdb.org/Study_340 to remove duplicates after double checking. We can perhaps find another way to remove duplicate studies since this is not ideal.

**Comment by lwaldron** [comment]
My understanding is that if you delete a study you need to either delete or move its experiments, or else they will be orphaned and then adopted by the next study anyone creates. Should these experiments be moved to another study or deleted @ftzohra22 ?

@tosfos what do you think about ensuring that new studies never reuse freed URLs? Ie a new study number is always Nmax + 1 where Nmax is the highest study number previously used? Then visiting a deleted study URL could somehow indicate that this study has been deleted. This display of orphaned experiments with an unrelated study is bad.

**Comment by lwaldron** [comment]
Now there are no experiments associated with https://bugsigdb.org/Study_340. @ftzohra22 did you delete or move them? Can you provide a list of relevant edit, delete, or move actions you've done? I don't recognize any clues in the Study 340 [history](https://bugsigdb.org/w/index.php?title=Study_340&action=history) so it's really bewildering to me what has been happening.

**Comment by ftzohra22** [comment]
These experiments needed to be completely deleted as they were just duplicates of another another study @lwaldron (https://bugsigdb.org/Study_95). For some reason, I'm guessing the experiments originally didn't get cleared when I was removing duplicates. I deleted all the experiments https://bugsigdb.org/Study_340 (deletion log: https://bugsigdb.org/Special:Log?type=delete&user=Fatima&page=&wpdate=&tagfilter=&subtype=) to avoid confusions for the new curator. They hadn't added any experiments yet.

**Comment by lwaldron** [comment]
I've confirmed that as soon as a study is deleted, its URL enters the front of the queue to be reused the next time a new study is created. I think this is bad behavior, the URL of a deleted study should never be reused during new study creation, it should just continue showing the "deleted study" message.

**Comment by tosfos** [comment]
Reserving Study numbers is not so simple. What do you think about using the PMID as a Study's unique ID? The URL of the study would be something like https://bugsigdb.org/30405010, but the Study's text title would be shown in most places (just like now). That would serve to ensure that nobody creates a study [with the same PMID as an existing study](https://bugsigdb.org/Project:Duplicate_studies). And it would also solve the issue you identified. Can we force all studies to have a PMID? If not, there might be a workaround for non-PMID studies.

Another thing we can think about is combining all Experiments and Signature pages directly onto the Study page instead of using subpages. There are advantages to having them separated out into subpages, but with the performance improvements, it's at least something we can think about. This would neatly keep all the Study/Experiment/Signature units together. Again, not a simple switch, and it can have significant ramifications that we need to think about, but if you have thoughts on it, let me know.

**Comment by lwaldron** [comment]
> What do you think about using the PMID as a Study's unique ID?

It's _so_ close to being a good solution - we have only a couple instances of desired non-PMID studies (https://bugsigdb.org/Study_562 and https://bugsigdb.org/Study_608), the rest have been mistakes by curators. Would it be possible to use DOI (although these contain a `/`), or the old-fashioned Study_N as a backup for non-PMID studies? If not I am still be tempted by the PMID as unique key option. The other valid use of two different studies with the same PMID because they have different study designs is very rare and dominated by mistaken duplication of the study instead of editing the existing study, so I think eliminating that use case would be a net benefit.

> Another thing we can think about is combining all Experiments and Signature pages directly onto the Study page instead of using subpages.

This sounds maybe too drastic for the scope of the problem right now at least, but good to keep in mind as a possibility. BTW we've had a bunch of new curators start in the past week as part an internship program, which is helping to identify issues, but it will probably calm down again in a couple weeks.

**Comment by tosfos** [comment]
I'm thinking a good compromise is to promote adding Studies by PMID only. We'll include instructions for how to add Studies by the old-fashioned Study_N (probably) for Studies that don't have a PMID. These non-PMID Studies will end up as being reusable if they are deleted, but I assume that will be rare enough to be acceptable.

We'll create a new form for adding studies by PMID. Once that looks good, we'll write a script to rename all the existing Studies that include a PMID to the new PMID naming system.

**Comment by lwaldron** [comment]
That sounds like an excellent compromise - normal users should have to intentionally go out of their way to create a study without a PMID. Except in the two unusual examples I gave, every instance so far of a user adding a non-PMID study has been an error (and there have been a couple dozen of those errors). When a study has a PMID, URI and DOI are unnecessary and a waste of time for users who think they need to provide them.

**Comment by lwaldron** [comment]
> Reserving Study numbers is not so simple. 

Just wondering if rather than reserving study numbers it would be straightforward to add a rule that creating a new study always uses N+1 where N is the largest existing study? The smaller and easy to remember study IDs are kind of handy and easy to remember, but not worth a lot of effort and I do want to eliminate reusing study IDs for different studies. Otherwise if we switch to PMID study numbers let's not change existing pages because there are already a lot of links to them.

**Comment by tosfos** [comment]
>  The smaller and easy to remember study IDs are kind of handy and easy to remember, but not worth a lot of effort and I do want to eliminate reusing study IDs for different studies.

It's a bit complicated, because we have to check for Studies that previously existed and then were deleted later so that we don't reuse them. We'll have to check how much effort is involved.

> Otherwise if we switch to PMID study numbers let's not change existing pages because there are already a lot of links to them.

I don't think this is a major concern. When we rename pages, it automatically leaves a redirect behind.

**Comment by lwaldron** [comment]
> It's a bit complicated, because we have to check for Studies that previously existed and then were deleted later so that we don't reuse them. We'll have to check how much effort is involved.

Is this only to protect against the corner case of a Study being created and then deleted before another Study with a larger number is added? That doesn't seem like an important corner case to me, what I don't like are 1) the problem of orphaned sub-pages getting assigned to a new Study and 2) having a study page that used to refer to one publication later refer to a different publication instead of just showing a "deleted page" message. Using PMID with a slightly hidden possibility of doing non-PMID studies does seem like a completely feasible solution with only a very slight loss of convenience of having smaller / more memorable study numbers currently.

**Comment by tosfos** [comment]
> Is this only to protect against the corner case of a Study being created and then deleted before another Study with a larger number is added?

Not just that case. I think it will happen if any Study is deleted (not just the most recent one). It will become the next number used when someone tries to add a new Study and that will cause the problems you mentioned:

> the problem of orphaned sub-pages getting assigned to a new Study

>  having a study page that used to refer to one publication later refer to a different publication instead of just showing a "deleted page" message.

We'll have to figure out the best option.

**Comment by tosfos** [comment]
We finally have a good solution for this. Is it OK to put the wiki in read-only mode for around 2 days to complete the conversion? @lwaldron

**Comment by lwaldron** [comment]
No problem!

**Comment by tosfos** [comment]
This was done. But there are some issues to address.

### Studies with Shared PMIDs
- [Study 10](https://bugsigdb.org/Study_10) / [Study 11](https://bugsigdb.org/Study_11) (Same PMID)  
- [Study 38](https://bugsigdb.org/Study_38) / [Study 39](https://bugsigdb.org/Study_39) (Same PMID)  
- [Study 82](https://bugsigdb.org/Study_82) / [Study 83](https://bugsigdb.org/Study_83) / [Study 84](https://bugsigdb.org/Study_84) (Same PMID)  
- [Study 971](https://bugsigdb.org/Study_971) / [Study 984](https://bugsigdb.org/Study_984) (Same PMID)  


### These studies have no PMID:
- [Study 1030](https://bugsigdb.org/Study_1030)  
- [Study 1078](https://bugsigdb.org/Study_1078)  
- [Study 1098](https://bugsigdb.org/Study_1098)  
- [Study 1112](https://bugsigdb.org/Study_1112)  
- [Study 1120](https://bugsigdb.org/Study_1120)  
- [Study 1123](https://bugsigdb.org/Study_1123)  
- [Study 1128](https://bugsigdb.org/Study_1128)  
- [Study 1129](https://bugsigdb.org/Study_1129)  
- [Study 1134](https://bugsigdb.org/Study_1134)  
- [Study 1135](https://bugsigdb.org/Study_1135)  
- [Study 1136](https://bugsigdb.org/Study_1136)  
- [Study 1137](https://bugsigdb.org/Study_1137)  
- [Study 1138](https://bugsigdb.org/Study_1138)  
- [Study 1147](https://bugsigdb.org/Study_1147)  
- [Study 1150](https://bugsigdb.org/Study_1150)  
- [Study 1153](https://bugsigdb.org/Study_1153)  
- [Study 1155](https://bugsigdb.org/Study_1155)  
- [Study 1164](https://bugsigdb.org/Study_1164)  
- [Study 1171](https://bugsigdb.org/Study_1171)  
- [Study 1199](https://bugsigdb.org/Study_1199)  
- [Study 1238](https://bugsigdb.org/Study_1238)  
- [Study 1272](https://bugsigdb.org/Study_1272)  
- [Study 1273](https://bugsigdb.org/Study_1273)  
- [Study 1274](https://bugsigdb.org/Study_1274)  
- [Study 1275](https://bugsigdb.org/Study_1275)  
- [Study 501](https://bugsigdb.org/Study_501)  
- [Study 513](https://bugsigdb.org/Study_513)  
- [Study 562](https://bugsigdb.org/Study_562)  
- [Study 608](https://bugsigdb.org/Study_608)  
- [Study 783](https://bugsigdb.org/Study_783)  
- [Study 817](https://bugsigdb.org/Study_817)  
- [Study 879](https://bugsigdb.org/Study_879)  
- [Study 887](https://bugsigdb.org/Study_887)  
- [Study 919](https://bugsigdb.org/Study_919)

**Comment by cmirzayi** [comment]
I like the change but we do occasionally have studies that don't have a PMID--either because they are preprints (though I believe NIH was piloting indexing preprints--who knows what's the future of that project in our current NIH context however) or because they are environmental microbiome studies that are not indexed in PubMed.

I know how messy it can be to have to have a second unique identifier but we need to consider how to maintain support for these studies.

**Comment by tosfos** [comment]
We added instructions for how to add a Study that has no PMID. Since this is occasional, it may be sufficient for editors to follow this. Please let us know if you require something more user-friendly.

[Help:How to add a Study](https://bugsigdb.org/w/index.php?title=Help:How_to_add_a_Study&type=revision&diff=188886&oldid=119650#Common_fields)

**Comment by lgeistlinger** [comment]
It might be good to take one step back here and carefully examine the downstream effects of the proposed change. Intermixing PMIDs and non-PMIDs as the primary bugsigdb study ID is not a clean solution and will cause a fair amount of headache when using the data downstream for analysis.

When we designed the data model we intentionally decided against using PMIDs as the primary study IDs as there are (i) studies that are not associated with a PMID, and (ii) PMIDs that have more than one associated study ID. There were good reasons to do so and I think we want to stick to that unless we'd like to abandon use cases (i) and (ii) entirely.

In general, my suggestion would be to be *very* conservative with such changes to the data model as they have the potential to introduce breaking changes to several packages that now depend on it downstream.

**Comment by cmirzayi** [comment]
@lgeistlinger Thank you for the insight. This came a bit too late as we did switch over to using PMIDs.

To that end @tosfos @lwaldron, we need a good solution for what to do for studies that do not have PMIDs. Right now, curators have been doing a workaround of using the DOI in place of the PMID. I don't have a good understanding of how unique identifiers work in the context of a SemanticWiki. Is there a way to have a fallback identifier like DOI formally? Or else any suggestions or thoughts on what we can do?

**Comment by lwaldron** [comment]
You can just leave the PMID blank and click "Create or Edit" study. PMID is not required.

**Comment by lwaldron** [comment]
> In general, my suggestion would be to be very conservative with such changes to the data model as they have the potential to introduce breaking changes to several packages that now depend on it downstream.

You're right and we should've had more discussion before implementing this, but it I don't think it's practical just to switch back now. Would the mapping list of old to new study IDs help for the downstream software? 

The use cases of a study without PMUD is still supported. The use case of two studies with one PMID has turned out to be exceedingly rare, and following a bunch of accidental repetitive curation of the same study, the wiki had already been enforcing unique PMID for some years without any requests for a valid request to repeat PMID for multiple studies, so that is a use case not worth supporting I think.

**Comment by cmirzayi** [comment]
Ok I'm fine with leaving PMID blank. We need to update the curation policy and study creation pages to provide this information as right now we've had multiple curators get "stuck" on this step because they weren't sure on what to do.

**Comment by cmirzayi** [comment]
I was able to modify the create study form to add the additional instruction that you can leave PMID blank.

![Image](https://github.com/user-attachments/assets/b4dfa8c7-1d5d-418e-8536-76ddfdaf36c3)

If you don't enter a PMID, you're taken to this page where you specify an identifier (i.e. the unique url component) for the study. Can we edit this to suggest that the curator use DOI? I don't have edit permissions for this page.

![Image](https://github.com/user-attachments/assets/32b29f6c-de11-4f69-b44f-1616a0ab46f2)

**Comment by lwaldron** [comment]
Tagging @feargalr because the change to naming studies by their PMID is going to break the vignette of their [TaxSEA](https://bioconductor.org/packages/TaxSEA) package (I'm sorry :( ). @jwokaty, would you open a new issue about the transition, and provide some guidance to developers of downstream packages and analyses?

@cmirzayi said:
> If you don't enter a PMID, you're taken to this page where you specify an identifier (i.e. the unique url component) for the study. Can we edit this to suggest that the curator use DOI? I don't have edit permissions for this page.

Just try to avoid the problem we've had before of one curator entering a PMID and another entering a DOI for the same study. I feel like we should only enter non-PubMed studies in special cases or urgently needed data, because someone should go back to update the data and the identifier once a preprint is published, and because even biorxiv preprints get a PMID now, so it has to be an obscure preprint server or something without an article to not have a PMID. I'm curious, what are the studies being entered that don't have a PMID?

**Comment by jwokaty** [comment]
@tosfos is it possible to correct this entry? And is it possible to have some error checking to prevent this in the future and add some examples on https://bugsigdb.org/Project:Add_or_edit_studies?

**Comment by cmirzayi** [comment]
We need to add some policies for valid study identifiers if there is no PMID. It's becoming a bit of a wild west unfortunately. I've observed several cases where the study identifier has been set as a full URL (which causes all sorts of problems) or they enter the title of the study which seems to affect the data exports (https://github.com/waldronlab/BugSigDB/issues/277).

Off the top of my head this page (https://bugsigdb.org/Special:FormStart?form=Study&page_name=), needs to enforce the following:

- No URLs
- No spaces if that affects exports?
- No punctuation if that affects exports?

We perhaps need to provide some sort of guidance on how to properly name the study in a way that creates a unique identifier while also being mostly fool-proof. I'm not sure if this is something the user comes up with or something we assign.

@lwaldron @jwokaty Any thoughts?

**Comment by jwokaty** [comment]
Will every paper have a DOI? If that's true, the `Start of form` should only accept the DOI and generate the ID from that. It could be what @cmirzayi did manually or `Study XX` since that format was previously used.

**Comment by seandavi** [comment]
For data modeling purposes, it might suffice to have a pmid field, a DOI field, each of which are constrained to be unique. As I think @jwokaty is alluding to, the ID of the study should be an internal one (bugsigdb_study_id, for example), again unique and assigned by the "system." I don't know what is technically required to create such a solution, though, nor do I have any sense of what ramifications such a change would have on the current system. So take my comments with a grain of salt.

**Comment by tosfos** [comment]
> My concern is that the "Just use the DOI as the PMID" workaround is not working well as BSDB was not designed with this in mind and it's causing issues. 

Yes, that field was not designed to handle a DOI, but we can change that.

> is it possible to correct this entry?

We will add handling for page names that include a slash.

> And is it possible to have some error checking to prevent this in the future and add some examples on https://bugsigdb.org/Project:Add_or_edit_studies?

We can basically enforce any restrictions on the page name. Should we make sure the input is in a valid PMID format? We can also have a separate field for creating a page using a DOI instead, and we can enforce that format in that field.

> Maybe just a random UUID, or a hash of what the curator enters?

Yes, we can have a separate button to create the page without a PMID or DOI which is more intuitive than having users leave it blank. And we can use Study XX like before and it increments each time. Alternatively, we can use a UUID but it's not so user-friendly.

> If this is rare maybe only PMID or DOI should be accepted and Start of form should only be accessible to admins for these rare occasions to generate a study ID, which I think should be made by bugsigdb.

We can do that.

I don't believe any of those changes will have significant ramifications on the existing system and how it operates, but we will verify once we know which implementation will give us the best user experience. 

There are other fancier options, so feel free to dream up a system and we'll figure out how much effort that will be. For example, we can do some dynamic checking on field values like DOI or PMID and issue a warning (or prevent saving) if there is a duplicate. Or we can do a lookup of the PMID or DOI as it's being typed and add a display that near the form input with a "not found" message until a valid input is there. And once the input is valid it can show the title of the study. That will encourage editors to put something valid there.

**Comment by tosfos** [comment]
That was one primary motivation. And there are other ways to solve it. Are you OK with us automatically deleting orphaned pages?

Another motivation was to avoid studies with duplicate PMIDs (the same study added twice). Is that still a factor? If it is, we can think of some way to warn users when they are doing that.

**Comment by cmirzayi** [comment]
@tosfos Yes, meeting with Levi right now and we are ok with automatically deleting orphaned pages if the parent study or experiment gets deleted.

I think we should disallow for multiple studies with the same PMID. If a person tries to do that, ideally it directs them to the study that's already there with that PMID.

**Comment by cmirzayi** [comment]
We now have a fun new issue where someone managed to set the study identifier to "NA" because that's what they entered as the PMID: https://bugsigdb.org/NA

I think the PMID field must be a well-formatted PMID (1-8 numeric digits with no leading 0s) or blank and nothing else. Wondering what your thoughts are @lwaldron.

**Comment by cmirzayi** [comment]
If you leave it blank you're taken to a page where you can manually name it. We've been telling them to use the DOI with no URL as the name.

The issue is once the PMID is set though we can't change it so that study is now and forever PMID: NA (until we delete it and recurate it).

**Comment by tosfos** [comment]
> I think the PMID field must be a well-formatted PMID

Should we enforce this on study creation?

> The issue is once the PMID is set though we can't change it so that study is now and forever PMID: NA (until we delete it and recurate it).

At least on the wiki side, we can help change any PMID. Please let us know if we can help.
